src/databricks/sql/thrift_backend.py

```diff
@@ -940,8 +954,11 @@ def fetch_results(
                     expected_row_start_offset, resp.results.startRowOffset
                 )
             )
+        are_arrow_results_compressed = (
+            resp.resultSetMetadata and resp.resultSetMetadata.lz4Compressed
```

Comment:
I don't think `resultSetMetadata` exists here. We'll have to capture it from the ExecuteStatement or GetOperationMetadata response and store it as cursor state.

Reply:
I think it exists as a field in response. Below is the class in ttypes.

class TFetchResultsResp(object):
    """
    Attributes:
     - status
     - hasMoreRows
     - results
     - resultSetMetadata
    """

I am not sure if you meant that this field won't be set in the response.

Reply:
This shortcut was added in Thrift V5. However, I don't think it is needed here as the `TResultSetMetadataResp` is already fetched at this point in `_results_message_to_execute_response`. We can capture the `lz4Compressed` field there and store it in `ResultSet` so it can be used here.

We had a discussion before to utilise this shortcut to save an extra roundtrip, but it would require more refactoring for not much value as it only is relevant for non-direct results.

Reply:
Done.